### PR TITLE
Add GTK4 input method modules

### PR DESCRIPTION
**Pull request by tinywrkb**
Add Fcitx5 and IBus GTK4 IM modules for East Asian languages input with the Ozone Wayland platform,
as we don't have yet support in Chromium for the relevant Wayland extensions, text-input-v3
and input-method-v2 popups.
GTK4 printing backends were added to match the packaged GTK3 printing backends.

To use this, GTK4 needed to be selected with the Chromium flag `--gtk-version=4`.

Relevant Chromium issues:
  - 1183262: Add support for gtk im module on ozone wayland platform: https://bugs.chromium.org/p/chromium/issues/detail?id=1183262
  - 1039161: Ozone-wayland: support text_input_v3 protocol: https://bugs.chromium.org/p/chromium/issues/detail?id=1039161

Two issues that are not mentioned in the commit message are popups with Fcitx5 IM module, and popup windows positioning.

Previously, popup windows were not showing up when using the Fcitx5 IM module.  
It seems like it's fixed now, not sure if the change was in Chromium or Fcitx5, but as far as I can tell, it's working correctly now. Tested with Chrome 98.0.4758.102 and Fcitx 5.0.14 (actually `4aa035547defca145092309e233913d509b5ff30`).  

If popups are not working for the user when running Fcitx5 server, then it's possible to use IBus IM module with the Fcitx5 IBus frontend.
Fcitx5's IBus frontend is enabled by default for the Fcitx5 server, so no extra setting is needed.
If Fcitx5 was globally set as the default IM module, then the user will need to add a Flatpak override `--env=GTK_IM_MODULE=IBus` for the app, and the IBus IM module will be used.

Relevant Chromium issues:
  - 1265118: Make Gtk 4 im module work with Fcitx 5: https://bugs.chromium.org/p/chromium/issues/detail?id=1265118
  - 1276059: IBus popup has wrong position with GTK4 IME (x11+wayland): https://bugs.chromium.org/p/chromium/issues/detail?id=1276059

### CODE REVIEW

**Comment by tinywrkb** [comment]
From what I can tell, default theme CSS generation with sassc is not needed.  
Also, I can work around the `gtk4-update-icon-cache` call in `build-aux/meson/post-install.py`, and avoid installing the hicolor theme icon.  
I don't see anything else I can remove or disable, it's pretty minimal now, and it's ready for review.

**Comment by tinywrkb** [comment]
I received an answer about this from @nanonyme on the matrix channel.  
The gist is that there are no plans to add GTK4 to the Freedesktop runtime currently, and apps that need it are expected to use the Gnome runtime.

So if we're adding here a GTK4 module, then we will need to keep maintaining it for quite a while.

It was also advised to bundle Pango and bump GTK4, so I've done that. With GTK 4.6 we can package LibAdwaita, but I'm not sure if it will be used, so I didn't add it.

**Comment by tinywrkb** [comment]
bot, build org.chromium.Chromium.BaseApp

**Comment by tinywrkb** [comment]
bot, build org.chromium.Chromium.BaseApp

**Comment by tinywrkb** [comment]
bot, build org.chromium.Chromium.BaseApp

**Comment by tinywrkb** [comment]
Latest commit bumps IBus to today's release (requires disabling systemd), and drops the IBus buildsystem hacks to avoid the hassle of maintaining them.  
Let me know if I should squash and remove the commented out sections in the packaging manifest.

**Comment by refi64** [comment]
Ah I guess this adds a curveball to my "only ship the Kerberos headers" thoughts :sweat_smile: 

It's unfortunate that fd.o doesn't seem to plan to ship the extension point, so I'll prioritize getting the flatpak-builder cleanup fix in & then looking over this in a bit more detail in the near future!

**Comment by tinywrkb** [comment]
I'm a bit conflicted about this.  
On the one hand, Ozone Wayland users who need to type in Chinese, Japanese...will want this.  

On the other hand, I don't like the idea of packaging GTK4. I was hoping that if we wait enough, then Chromium's devs will add support for the Wayland extensions, and maybe even for GTK3 IM modules.

I guess this can be kept on hold as long as Chromium doesn't set Wayland as the default Ozone platform, but after that there should be some support for IME frameworks.

**Comment by tinywrkb** [comment]
Rebased and updated.

**Comment by tinywrkb** [comment]
@OctopusET [revived the text_input_manager_v3 patch set for Chromium](https://chromium-review.googlesource.com/c/chromium/src/+/3750452).  
I support waiting to see how that effort will pan out, and for the meantime, keep this PR on hold.

A possible workaround for users that want the GTK4 IM modules working with Ozone Wayland is to run with the Gnome runtime.  
We actually don't have popups/candidate selection, and that's because of the outdated Fcitx5 module (`5.0.10`) in Gnome 42, but Gnome 43 will have version `5.0.14`, so I guess it should be fine after making the switch here to Freedesktop 22.08, and Gnome 43 will be out.  
And I see that we have in Gnome the CUPS backend, so the user will lose nothing.  
I'm not quite sure why the IBus IM module doesn't work at all, maybe it also too dated? I haven't really looked into it.  

So, basically, you should run the browser like this:

```
$ flatpak run --runtime=org.gnome.Platform//43 com.google.Chrome
```

You may or may not need to add `--unset-env=GTK_PATH`. I didn't need this with Fcitx5 and Gnome 43rc.

And of course, you still need to add the `--gtk-version=4` flag and select the Ozone Wayland platform.

**Comment by tinywrkb** [comment]
Updated for 22.08, also bumped all the added modules, and included the suggested CUPS print backend removal.

Seeing that changing the runtime to Gnome 43 works now perfectly with any of the IM modules, Fcitx5 and IBus, and that [the patchset for Chromium](https://chromium-review.googlesource.com/c/chromium/src/+/3750452) has been revived, I'm less inclined to push this forward.  

My previous opinion about this still stands. As long as Wayland is not set as the default Ozone platform, we should wait on Wayland IME extensions support and keep GTK4 out of this base application. Hopefully, Google/Igalia won't force our hand to throw this in.

**Comment by bbhtt** [comment]
This needs to be picked up sometime soon considering https://chromium-review.googlesource.com/c/chromium/src/+/6310469 is coming. 

I'll probably schedule this for 25.08 and leave 24.08 alone (Same for electron).
